src/Client/Web/Bit.Client.Web.BlazorUI/Components/Buttons/BitToggleButton.razor.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Web;
+
+namespace Bit.Client.Web.BlazorUI
+{
+    public partial class BitToggleButton
+    {
+        private ButtonStyle buttonStyle = ButtonStyle.Primary;
+
+        private int? tabIndex;
+        private bool isChecked;
+
+        [Parameter] public bool AllowDisabledFocus { get; set; } = true;
+        [Parameter] public string? AriaDescription { get; set; }
+        [Parameter] public bool AriaHidden { get; set; }
+        [Parameter] public string? AriaLabel { get; set; }
+        /// <summary>
+        /// determine if the button is checked state, default is true
+        /// </summary>        
+        [Parameter]
+        public bool IsChecked
+        {
+            get => isChecked;
+            set
+            {
+                if (value == isChecked) return;
+                isChecked = value;
+                ClassBuilder.Reset();
+                _ = IsCheckedChanged.InvokeAsync(value);
+            }
+        }
+        [Parameter] public EventCallback<bool> IsCheckedChanged { get; set; }
+        /// <summary>
+        /// the icon that shows in checked state
+        /// </summary>
+        [Parameter] public string? CheckedIconName { get; set; }
+        /// <summary>
+        /// the icon that shows in unChecked state
+        /// </summary>
+        [Parameter] public string? UnCheckedIconName { get; set; }
+        /// <summary>
+        /// the text that shows in Checked state
+        /// </summary>
+        [Parameter] public string? ChekedLabel { get; set; }
+        /// <summary>
+        /// the text that shows in unChecked state
+        /// </summary>
+        [Parameter] public string? UnChekedLabel { get; set; }
```

Comment:
there is no need to introduce 2 different parameters for the Label. just define a single parameter named Label, it is the consumer's job to define the correct label text based on the current state of the IsChecked parameter.